apps/prairielearn/src/ee/pages/instructorAiGenerateQuestion/instructorAiGenerateQuestion.ts

```diff
@@ -137,14 +142,25 @@ router.post(
         );
       }
 
+      const qid = genJobs[0]?.data['questionQid'];
+
+      const threads: AiGenerationPrompt[] = await queryRows(
+        sql.select_generation_thread_items,
+        { qid, course_id: res.locals.course.id.toString() },
+        AiGenerationPromptSchema,
+      );
+
       const result = await regenerateQuestion(
         client,
         res.locals.course.id,
         res.locals.authn_user.user_id,
-        genJobs[0]?.data?.prompt,
+        threads[0]?.user_prompt,
         req.body.prompt,
-        genJobs[0]?.data?.html,
-        genJobs[0]?.data?.python,
+        threads[threads.length - 1].html || '',
+        threads[threads.length - 1].python || '',
```

Comment:
Checking my understanding: we're using the `user_prompt` from the first item and `html`/`python` from the last one because with the current UI, we always use the original prompt when doing revisions, but revisions are always done based on whatever HTML/Python the most recent generation produced, is that right?

Reply:
yeah the prompts are sorted in order, so we want the *original* prompt and the *final* html/python generation.